modules/system/classes/CompilableAssets.php

```diff
@@ -0,0 +1,340 @@
+                } catch (SystemException $e) {
+                    // Either the package name or the config file path have already been registered, skip.
+                    continue;
+                }
+            }
+        }
+    }
+
+    /**
+     * Register a compilable config.
+     */
+    public function registerCompilable(string $name, array $config): void
+    {
+        $this->compilableConfigs[$name] = $config;
+    }
+
+    /**
+     * Registers a callback for processing.
+     */
+    public static function registerCallback(callable $callback): void
+    {
+        static::$callbacks[] = $callback;
+    }
+
+    /**
+     * Calls the deferred callbacks.
+     */
+    public function fireCallbacks(): void
+    {
+        // Call callbacks
+        foreach (static::$callbacks as $callback) {
+            $callback($this);
+        }
+    }
+
+    /**
+     * Returns the count of packages registered.
+     */
+    public function getPackageCount(): int
+    {
+        return array_sum(array_map(fn ($packages) => count($packages), ...$this->packages));
+    }
+
+    /**
+     * Returns all packages registered.
+     */
+    public function getPackages(string $type, bool $includeIgnored = false): array
+    {
+        $packages = $this->packages[$type] ?? [];
+
+        ksort($packages);
+
+        if (!$includeIgnored) {
+            return array_filter($packages, function ($package) {
+                return !($package['ignored'] ?? false);
+            });
+        }
+
+        return $packages;
+    }
+
+    /**
+     * Returns if package(s) is registered.
+     */
+    public function hasPackage(string $name, bool $includeIgnored = false): bool
+    {
+        foreach ($this->packages ?? [] as $packages) {
+            foreach ($packages as $packageName => $config) {
+                if (($name === $packageName) && (!$config['ignored'] || $includeIgnored)) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns package(s).
+     */
+    public function getPackage(string $name, bool $includeIgnored = false): array
+    {
+        $results = [];
+        foreach ($this->packages ?? [] as $packages) {
+            foreach ($packages as $packageName => $config) {
+                if (($name === $packageName) && (!$config['ignored'] || $includeIgnored)) {
+                    $results[] = $config;
+                }
+            }
+        }
+
+        return $results;
+    }
+
+    /**
+     * Registers an entity as a package for compilation.
+     *
+     * Entities can include plugins, components, themes, modules and much more.
+     *
+     * The name of the package is an alias that can be used to reference this package in other methods within this
+     * class.
+     *
+     * By default, the CompilableAssets class will look for a `package.json` file for Node dependencies, and a config
```

Comment:
Always use code blocks around names, as that way people using translation tools won't inadvertently translate a named entity, unless it's within a docblock tag.
```suggestion
     * By default, the `CompilableAssets` class will look for a `package.json` file for Node dependencies, and a config
```